### PR TITLE
Doc: Documentation for --deploy extension and few extension differences

### PR DESCRIPTION
**Pull request by tapaswenipathak**
### The issue

Fixes https://github.com/pypa/pipenv/issues/3383.

### The fix

Added documental information for the extension.

### The checklist

* [X] Associated issue
* [X] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix`, `.feature`, `.behavior`, `.doc`. `.vendor`. or `.trivial` (this will appear in the release changelog).

### CODE REVIEW

File: docs/basics.rst [hunk 1]
```diff
@@ -315,6 +315,7 @@ The user can provide these additional parameters:
 
     - ``--dev`` — Install both ``develop`` and ``default`` packages from ``Pipfile``.
     - ``--system`` — Use the system ``pip`` command rather than the one from your virtualenv.
+    - ``--deploy --system`` — The workflow is using pipenv for package loc in local development environment for having a Pipfile and Pipfile.locl file. Deployment can be done using the command. The ``--deploy`` makes sure the packages are properly locked in Pipfile.lock.
```
Comment (frostming) [inline docs/basics.rst]: Hmmm, I prefer to mention `--deploy` option only, not a combination, to make them consistent.

File: docs/basics.rst [hunk 2]
```diff
@@ -404,3 +405,11 @@ production environments for reproducible builds.
     If you'd like a ``requirements.txt`` output of the lockfile, run ``$ pipenv lock -r``.
     This will include all hashes, however (which is great!). To get a ``requirements.txt``
     without hashes, use ``$ pipenv run pip freeze``.
+
+
+☤ Few notable differences
+-------------------------
+
+- ``pip install pipenv`` is installing pipenv as a tool.
```
Comment (frostming) [inline docs/basics.rst]: Why mention `pip install pipenv` here, `pip` is a different tool from `pipenv`.

File: docs/basics.rst [hunk 3]
```diff
@@ -404,3 +405,11 @@ production environments for reproducible builds.
...
+- ``pip install --system`` installs packages in the machine.
```
Comment (frostming) [inline docs/basics.rst]: `pip` or `pipenv`?

File: docs/basics.rst [hunk 4]
```diff
@@ -404,3 +405,11 @@ production environments for reproducible builds.
...
+- ``pipenv install`` will create virtualenv and install package in it.
```
Comment (frostming) [inline docs/basics.rst]: No need to mention `pipenv install` again given it is described in CLI docs and quick start

File: docs/basics.rst (hunk 2 above)
Comment (tapaswenipathak) [reply docs/basics.rst]: The statements are useful as the users are confused between these options and maintainers feel documentation should define the items properly. Makes sense?

Comment (frostming) [reply docs/basics.rst]: Clarifying the differences between these commands is valuable. We already have some notes in https://docs.pipenv.org/en/latest/advanced/#using-pipenv-for-deployments but it is way too simple. You should add more doc there about the differences between `pipenv sync`, `pipenv install --deploy`, `pipenv install --ignore-pipfile` and other similar commands, and what is recommended under particular situation.

Comment (frostming) [reply docs/basics.rst]: I would advise you remove this section here.

Comment (tapaswenipathak) [reply docs/basics.rst]: I reffing the comment https://github.com/pypa/pipenv/pull/3939#discussion_r322034530 in the ticket I created.

**Comment by tapaswenipathak** [comment]
Hey @frostming: Done per review, sorry I missed the review comment.

File: docs/basics.rst [hunk 5]
```diff
@@ -321,7 +321,7 @@ The user can provide these additional parameters:
 
     - ``--dev`` — Install both ``develop`` and ``default`` packages from ``Pipfile``.
     - ``--system`` — Use the system ``pip`` command rather than the one from your virtualenv.
-    - ``--deploy`` — The workflow is using pipenv for package loc in local development environment for having a Pipfile and Pipfile.lock. The ``--deploy`` makes sure the packages are properly locked in Pipfile.lock.
+    - ``--deploy`` — The workflow is using pipenv for package loc in local development environment for having a Pipfile and Pipfile.lock. The ``--deploy`` makes sure the packages are properly locked in Pipfile.lock, and abort if the lock file is out-of-date.
```
Comment (frostming) [inline docs/basics.rst]: The first sentence is meaningless, To be consistent, just
`— Make sure the packages are properly locked in Pipfile.lock, and abort if the lock file is out-of-date.` is enough

**Comment by tapaswenipathak** [comment]
@frostming: merge now?

**Comment by tapaswenipathak** [comment]
@frostming: anything remaining?

**Comment by tapaswenipathak** [comment]
@frostming: merge ping.

**Pull request merged**